File: samps/src/practice.java
import java.io.File;
import java.io.RandomAccessFile;

/**
 * Practice program, sigma positive integers.
 * 
 * @author pc2@ecs.csus.edu
 * @version $Id$
 */

// $HeadURL$
public class practice {

    /**
     * Get Current Working Directory.
     * 
     * @return current working directory.
     */
    private static String getCurrentDirectory() {
        File curdir = new File(".");

        try {
            return curdir.getCanonicalPath();
        } catch (Exception e) {
            // ignore exception
            return ".";
        }
    }

    public static void main(String[] args) {
        String filename = "practice.dat";
        try {

            RandomAccessFile file = new RandomAccessFile(filename, "r");
            String line;

            int inval = 0;
            int sum = 0;

            while ((line = file.readLine()) != null) {
                inval = new Integer(line.trim()).intValue();
                sum = 0;

                if (inval == 0)
                    System.exit(0);

                if (inval < 1) {
                    for (int i = 1; i >= inval; i--)
                        sum += i;
                } else {
                    for (int i = 1; i <= inval; i++)
                        sum += i;
                }
                System.out.println("N = " + inval + "  Sum = " + sum);
            }

            System.out.println("Did not find trailing zero");
        } catch (Exception e) {
            System.out.println("Possible trouble reading " + filename + " in " + getCurrentDirectory());
        }
        System.exit(4);
    }
}

// eof practice.java $Id$

File: samps/src/practice.cpp
//
// File:    practice.cpp
// Purpose: to sum from the input number to zero 
// Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
//
// caveat - this is not nice code, copy at own risk ;)
//
// $Id$
//
// $Header$
//
//

#include <fstream>
#include <iostream>
using namespace std;

main ()
{
	int num;
	int sum;
	int i;
	ifstream filein("practice.dat");

	if (filein.fail())
	{
		cerr << "Could not read from file practice.dat " << endl;
	}
	else
	{

		sum = 0;

		while(!filein.eof())
		{
			filein >> num;
			sum = 0;

			if (num == 0)
				exit (0);

			if (num < 1)
			{
				for (i = 1; i >= num; i --)
				       sum += i;	
			}
			else
			{
				for (i = 1; i <= num; i ++)
				       sum += i;	
			}
			// cout << "N = " << num << " Sum = " << sum << endl;
			// requires formatting
		  printf("N = %-3d    Sum = %d\n", num, sum);

		}

		cout << "Did not find trailing zero " << endl;
	}
}

// eof $Id$ practice.cpp
